pkg/cortex/modules.go

```diff
@@ -298,7 +298,14 @@ func (t *Cortex) initTableManager(cfg *Config) error {
 		return err
 	}
 
-	t.tableManager, err = chunk.NewTableManager(cfg.TableManager, cfg.Schema, cfg.Ingester.MaxChunkAge, tableClient)
+	var bucketClient chunk.BucketClient
+	if cfg.Storage.FSConfig.Directory != "" {
```

Comment:
I feel like this logic is misplaced a will become brittle.  Could we always call NewBucketClient here, and have the big switch in that function no when to return nil and when to return an error?